src/Microsoft.Azure.WebJobs.Host/Executors/FunctionExecutor.cs

```diff
@@ -92,19 +87,15 @@ private async Task<IDelayedException> TryExecuteAsyncCore(IFunctionInstanceEx fu
                 {
                     try
                     {
-                        instanceLogEntry = await NotifyPreBindAsync(functionStartedMessage);
```

Comment:
The pattern repeated many times below is that we have a small async helper function that does some non-async work then goes async. Where it makes sense and keeps the code readable, I've broken these up.